monkey/tests/unit_tests/monkey_island/cc/repository/test_mongo_agent_repository.py

```diff
@@ -172,6 +187,24 @@ def test_get_running_agents__retrieval_error(error_raising_agent_repository):
         error_raising_agent_repository.get_running_agents()
 
 
+@pytest.mark.parametrize("actual_agent", [DESCENDANT_AGENT, PROGENITOR_AGENT])
+def test_get_progenitor(agent_repository, actual_agent):
+    expected_progenitor = agent_repository.get_progenitor(actual_agent)
+
+    assert expected_progenitor == PROGENITOR_AGENT
```

Comment:
`PROGENITOR_AGENT` is the expected progenitor. The `actual_progenitor` is the one that `get_progenitor` calculates. `actual_agent` could be better named. Probably just `agent`.